test/unit/cloudantBaseService.test.js

```diff
@@ -166,4 +167,102 @@ describe('Test CloudantBaseService', () => {
     assert.ok(service.baseOptions.jar === undefined);
     assert.strictEqual(service.getAuthenticator().tokenManager.url, iamUrl);
   });
+
+  describe('Check keepAlive', () => {
+    const auth = new IamAuthenticator({
+      apikey: '1234',
+      clientId: '345',
+      clientSecret: '234',
+    });
+
+    const agentOptions = {
+      keepAlive: true,
+      maxSockets: 6,
+    };
+
+    function assertDefaultAgentOptions(service, agents) {
+      Object.keys(agentOptions).forEach((agentOption) => {
+        Object.values(agents).forEach((agent) => {
+          assert.strictEqual(
+            service.baseOptions[agent][agentOption],
+            agentOptions[agentOption]
+          );
+        });
+      });
+    }
+    describe('When it is set in the constructor', () => {
```

Comment:
```suggestion
    describe('Use service default Agent', () => {
```

Reply:
I removed the `describe` blocks and used better namings in the `it` blocks in 69652b5  and c55694c